network/builder.go

```diff
@@ -13,24 +13,100 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	defaultAddress = "tcp://localhost:8588"
+)
+
+var (
+	// ErrDuplicatePlugin returns if the plugin has already been registered
+	// with the builder
+	ErrDuplicatePlugin = errors.New("builder: plugin is already registered")
+	// ErrNoAddress returns if no address was given to the builder
+	ErrNoAddress = errors.New("builder: network requires public server IP for peers to connect to")
+	// ErrNoKeyPair returns if no keypair was given to the builder
+	ErrNoKeyPair = errors.New("builder: cryptography keys not provided to Network; cannot create node ID")
```

Comment:
May be these should be strings instead of errors. Then you can do errors.New(errNoKeyPair) when the error actually happened. I think this model was from a Java thing

Reply:
this is pretty standard though, you see this a lot in open source go. it's only weird now because i haven't seen pkg/errors until now

Reply:
I think with the go "errors" package, it's pretty much a string with no stack, so it didn't matter.

Reply:
We should standardize with the pkg/errors model.